### PR TITLE
feat: Allow adding `<properties>` to `<testcase>`

### PR DESCRIPTION
**Pull request by eps1lon**
Closes https://github.com/jest-community/jest-junit/issues/246

Create a file in your project root directory named junitTestCaseProperties.js:
```js
module.exports = (testResult) => {
  return {
    "dd_tags[test.invocations]": testResult.invocations,
  };
};
```

Will render
```xml
<testsuites name="jest tests">
  <testsuite name="addition" tests="1" errors="0" failures="0" skipped="0" timestamp="2017-07-13T09:42:28" time="0.161">
    <testcase classname="addition positive numbers should add up" name="addition positive numbers should add up" time="0.004">
      <properties>
        <property name="dd_tags[test.invocations]" value="1" />
      </properties>
    </testcase>
  </testsuite>
</testsuites>
```

WARNING: Properties for testcases is not following standard JUnit XML schema.
However, other consumers may support properties for testcases like [DataDog metadata through `<property>` elements](https://docs.datadoghq.com/continuous_integration/tests/junit_upload/?tab=jenkins#providing-metadata-through-property-elements)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,4 +256,32 @@ Will render
 </testsuites>
 ```
 
+#### Adding custom testcase properties
+New feature as of jest-junit 11.0.0!
```
Comment (eps1lon) [inline README.md:260]:
```suggestion
```

**Review by AfefZn — APPROVED** [review]
It works correctly!

**Comment by AfefZn** [comment]
Hello @palmerj3, Could You please review this pull request?

**Comment by AfefZn** [comment]
@SimenB, Could You please review this pull request?

**Review by palmerj3 — APPROVED** [review]

**Comment by palmerj3** [comment]
Looks great!

Sorry for the delay in reviewing. Work has been very busy lately.

**Comment by palmerj3** [comment]
This has now been published in jest-junit v16 https://github.com/jest-community/jest-junit/releases/tag/v16.0.0

**Pull request merged**